WordPress/src/main/java/org/wordpress/android/ui/people/PeopleAdapter.java

```diff
@@ -8,45 +8,50 @@
 import android.widget.TextView;
 
 import org.wordpress.android.R;
+import org.wordpress.android.datasets.PeopleTable;
 import org.wordpress.android.models.Person;
 import org.wordpress.android.models.Role;
 import org.wordpress.android.util.GravatarUtils;
 import org.wordpress.android.widgets.WPNetworkImageView;
 
-import java.util.ArrayList;
 import java.util.List;
 
 public class PeopleAdapter extends BaseAdapter {
     private final Context mContext;
     private final LayoutInflater mInflater;
-    private List<Person> mPersonList;
+    private List<Person> mPeopleList;
     private int mAvatarSz;
+    private int mLocalBlogId;
 
-    public PeopleAdapter(Context context) {
+    public PeopleAdapter(Context context, int localBlogId) {
         mContext = context;
         mAvatarSz = context.getResources().getDimensionPixelSize(R.dimen.avatar_sz_medium);
         mInflater = LayoutInflater.from(context);
-        mPersonList = new ArrayList<>();
-        mPersonList.add(new Person(1, "beaulebens", "Beau", "Lebens", "Beau", "http://lorempixum.com/76/76", Role.ADMIN));
-        mPersonList.add(new Person(2, "ebinnion", "Eric", "Binnion", "Eric", "http://lorempixum.com/76/76", Role.AUTHOR));
-        mPersonList.add(new Person(3, "javialvarez", "Javi", "Alvarez", "Javi", "http://lorempixum.com/76/76", Role.CONTRIBUTOR));
-        mPersonList.add(new Person(4, "oguzkocer", "Oguz", "Kocer", "Oguz", "http://lorempixum.com/76/76", Role.EDITOR));
+        mLocalBlogId = localBlogId;
+        mPeopleList = PeopleTable.getPeople(mLocalBlogId);
+    }
+
+    @Override
+    public void notifyDataSetChanged() {
+        mPeopleList = PeopleTable.getPeople(mLocalBlogId);
```

Comment:
I think that this override makes the code more complex and "dangerous" than is needed, since it breaks the [notifyDataSetChanged]'s documentation (http://developer.android.com/intl/es/reference/android/widget/BaseAdapter.html#notifyDataSetChanged()) by mutating the Adapter instead of just notifying the observers. The caller should assume that the adapter's data have already been changed by the time `notifyDataChanged()` is called but this override invalidates that assumption.

I think the data should be _provided_ to the Adapter (via a constructor or setter) and if needed as a convenience, reloaded by a helper method (say, "reloadFromDB(localBlogId)" for example) whose explicit function would be to load the list from the DB. The local mLocalBlogId  variable would not even be needed then.
The (unoverridden) `notifyDataSetChanged` could then be called by that helper or be left to the helper's caller to do it.